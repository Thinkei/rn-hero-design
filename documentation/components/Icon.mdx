---
name: Icon
menu: Components
---

import { Playground } from 'docz'
import { Props } from 'gatsby-theme-docz/src/components/Props';
import { Icon } from 'rn-hero-design'
import IconList from './IconList'

# Icon

<IconList
  icons={[
    'email',
    'eye',
    'eye-invisible',
    'ok-circle',
    'calendar',
    'more-vertical',
  ]} 
/>
<IconList
  icons={[
    'email-outline',
    'eye-outline',
    'eye-invisible-outline',
    'cancel-outline',
    'clock-circle-outline',
    'comment-outline',
    'plus-circle-outline',
    'plus-outline',
    'piggy-bank-outline',
    'target-outline',
    'single-right-outline',
    'single-left-outline',
    'calendar-outline',
    'phone-outline',
    'face-id',
    'arrow-down',
    'camera-outline',
    'paperclip',
    'image-outline',
  ]} 
/>

## Basic usage

<Playground>
  <Icon icon="email" />
</Playground>

## Properties

<Props props_={{
  icon: { type: 'String', required: true },
  size: { type: 'Number', required: false, defaultValue: { value: '24' } },
  color: { type: 'String', required: false },
  wrapperStyle: { type: 'Object', required: false },
  theme: { type: 'Object', required: false, description: 'The theme object contains style of all components. Can be injected by `injectTheme` HOC' },
}} />

