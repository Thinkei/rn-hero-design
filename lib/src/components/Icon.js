import React from 'react';
import t from 'prop-types';
import { View, StyleSheet } from 'react-native';
import { SvgXml } from 'react-native-svg';
import heroTheme from '../themes/hero';

import email from '../icons/email.svg';
import emailOutline from '../icons/email-outline.svg';
import eye from '../icons/eye.svg';
import eyeOutline from '../icons/eye-outline.svg';
import eyeInvisible from '../icons/eye-invisible.svg';
import eyeInvisibleOutline from '../icons/eye-invisible-outline.svg';
import okCircle from '../icons/ok-circle.svg';
import calendar from '../icons/calendar.svg';
import clockCircleOutline from '../icons/clock-circle-outline.svg';
import commentOutline from '../icons/comment-outline.svg';
import cancelOutline from '../icons/cancel-outline.svg';
import plusCircleOutline from '../icons/plus-circle-outline.svg';
import plusOutline from '../icons/plus-outline.svg';

const SVG_ICONS = {
  email,
  'email-outline': emailOutline,
  eye,
  'eye-outline': eyeOutline,
  'eye-invisible': eyeInvisible,
  'eye-invisible-outline': eyeInvisibleOutline,
  'ok-circle': okCircle,
  calendar,
  'clock-circle-outline': clockCircleOutline,
  'comment-outline': commentOutline,
  'cancel-outline': cancelOutline,
  'plus-circle-outline': plusCircleOutline,
  'plus-outline': plusOutline,
};

const Icon = ({
  icon,
  size = 24,
  color,
  wrapperStyle,
  theme = heroTheme,
  ...props
}) =>
  SVG_ICONS[icon] ? (
    <View
      style={StyleSheet.compose(
        {
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
          width: size,
          height: size,
        },
        wrapperStyle,
      )}
      {...props}
    >
      <SvgXml
        xml={SVG_ICONS[icon]}
        override={{
          width: size,
          height: size,
          fill: color || theme.icon.icon.color,
        }}
      />
    </View>
  ) : null;

Icon.propTypes = {
  /**
   * Name of the icon, in kebab case
   */
  icon: t.string,
  size: t.number,
  color: t.string,
  wrapperStyle: t.object,
  /**
   * The theme object contains style of all components. Can be injected by `injectTheme` HOC
   */
  theme: t.object,
};

export default Icon;
