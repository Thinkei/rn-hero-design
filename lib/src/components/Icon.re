open ReactNative;

[@bs.get] external getColorProperty: Style.t => Color.t = "color";

module SvgXml = {
  [@react.component] [@bs.module "react-native-svg"]
  external make: (~xml: string, ~override: Js.t('a)) => React.element =
    "SvgXml";
};

[@bs.module "../icons/email"] external email: string = "default";
[@bs.module "../icons/email-outline"]
external emailOutline: string = "default";
[@bs.module "../icons/eye"] external eye: string = "default";
[@bs.module "../icons/eye-outline"] external eyeOutline: string = "default";
[@bs.module "../icons/eye-invisible"]
external eyeInvisible: string = "default";
[@bs.module "../icons/eye-invisible-outline"]
external eyeInvisibleOutline: string = "default";
[@bs.module "../icons/ok-circle"] external okCircle: string = "default";
[@bs.module "../icons/calendar"] external calendar: string = "default";
[@bs.module "../icons/clock-circle-outline"]
external clockCircleOutline: string = "default";
[@bs.module "../icons/comment-outline"]
external commentOutline: string = "default";
[@bs.module "../icons/cancel-outline"]
external cancelOutline: string = "default";
[@bs.module "../icons/plus-circle-outline"]
external plusCircleOutline: string = "default";
[@bs.module "../icons/plus-outline"] external plusOutline: string = "default";
[@bs.module "../icons/piggy-bank-outline"]
external piggyBankOutline: string = "default";
[@bs.module "../icons/target-outline"]
external targetOutline: string = "default";
[@bs.module "../icons/single-right-outline"]
external singleRightOutline: string = "default";
[@bs.module "../icons/more-vertical"]
external moreVertical: string = "default";
[@bs.module "../icons/calendar-outline"]
external calendarOutline: string = "default";
[@bs.module "../icons/phone-outline"]
external phoneOutline: string = "default";
[@bs.module "../icons/face-id"] external faceId: string = "default";
[@bs.module "../icons/single-left-outline"]
external singleLeftOutline: string = "default";
[@bs.module "../icons/arrow-down"] external arrowDown: string = "default";
[@bs.module "../icons/arrow-up"] external arrowUp: string = "default";
[@bs.module "../icons/camera-outline"]
external cameraOutline: string = "default";
[@bs.module "../icons/paperclip"] external paperclip: string = "default";
[@bs.module "../icons/image-outline"]
external imageOutline: string = "default";
[@bs.module "../icons/file-outline"] external fileOutline: string = "default";
[@bs.module "../icons/warning-circle-outline"]
external warningCircleOutline: string = "default";
[@bs.module "../icons/warning-circle"]
external warningCircle: string = "default";
[@bs.module "../icons/radio-active"] external radioActive: string = "default";
[@bs.module "../icons/radio-inactive"]
external radioInactive: string = "default";
[@bs.module "../icons/plus-circle"] external plusCircle: string = "default";
[@bs.module "../icons/cancel-circle"]
external cancelCircle: string = "default";
[@bs.module "../icons/cancel-circle-outline"]
external cancelCircleOutline: string = "default";
[@bs.module "../icons/ok-circle-outline"]
external okCircleOutline: string = "default";
[@bs.module "../icons/folder"] external folder: string = "default";
[@bs.module "../icons/folder-outline"]
external folderOutline: string = "default";
[@bs.module "../icons/stopwatch"] external stopwatch: string = "default";
[@bs.module "../icons/stopwatch-outline"]
external stopwatchOutline: string = "default";
[@bs.module "../icons/globe"] external globe: string = "default";
[@bs.module "../icons/globe-outline"]
external globeOutline: string = "default";
[@bs.module "../icons/suitcase"] external suitcase: string = "default";
[@bs.module "../icons/suitcase-outline"]
external suitcaseOutline: string = "default";
[@bs.module "../icons/level"] external level: string = "default";
[@bs.module "../icons/level-outline"]
external levelOutline: string = "default";

[@bs.module "../icons/format_bold"] external formatBold: string = "default";
[@bs.module "../icons/format_italic"]
external formatItalic: string = "default";
[@bs.module "../icons/format_underlined"]
external formatUnderlined: string = "default";
[@bs.module "../icons/format_list_bulleted"]
external formatListBulleted: string = "default";
[@bs.module "../icons/format_list_numbered"]
external formatListNumbered: string = "default";

[@bs.module "../icons/share-1.js"] external share1: string = "default";
[@bs.module "../icons/share-2.js"] external share2: string = "default";
[@bs.module "../icons/browser-outline.js"]
external browserOutline: string = "default";
[@bs.module "../icons/restart-outline"]
external restartOutline: string = "default";

let xmlFromIcon = icon =>
  switch (icon) {
  | "email" => Some(email)
  | "email-outline" => Some(emailOutline)
  | "eye" => Some(eye)
  | "eye-outline" => Some(eyeOutline)
  | "eye-invisible" => Some(eyeInvisible)
  | "eye-invisible-outline" => Some(eyeInvisibleOutline)
  | "ok-circle" => Some(okCircle)
  | "calendar" => Some(calendar)
  | "calendar-outline" => Some(calendarOutline)
  | "clock-circle-outline" => Some(clockCircleOutline)
  | "comment-outline" => Some(commentOutline)
  | "cancel-outline" => Some(cancelOutline)
  | "plus-circle-outline" => Some(plusCircleOutline)
  | "plus-outline" => Some(plusOutline)
  | "piggy-bank-outline" => Some(piggyBankOutline)
  | "target-outline" => Some(targetOutline)
  | "single-right-outline" => Some(singleRightOutline)
  | "single-left-outline" => Some(singleLeftOutline)
  | "more-vertical" => Some(moreVertical)
  | "phone-outline" => Some(phoneOutline)
  | "face-id" => Some(faceId)
  | "arrow-down" => Some(arrowDown)
  | "arrow-up" => Some(arrowUp)
  | "camera-outline" => Some(cameraOutline)
  | "paperclip" => Some(paperclip)
  | "image-outline" => Some(imageOutline)
  | "file-outline" => Some(fileOutline)
  | "warning-circle-outline" => Some(warningCircleOutline)
  | "warning-circle" => Some(warningCircle)
  | "radio-active" => Some(radioActive)
  | "radio-inactive" => Some(radioInactive)
  | "format_bold" => Some(formatBold)
  | "format_italic" => Some(formatItalic)
  | "format_underlined" => Some(formatUnderlined)
  | "format_list_numbered" => Some(formatListNumbered)
  | "format_list_bulleted" => Some(formatListBulleted)
  | "plus-circle" => Some(plusCircle)
  | "cancel-circle" => Some(cancelCircle)
  | "cancel-circle-outline" => Some(cancelCircleOutline)
  | "ok-circle-outline" => Some(okCircleOutline)
  | "folder" => Some(folder)
  | "folder-outline" => Some(folderOutline)
  | "stopwatch" => Some(stopwatch)
  | "stopwatch-outline" => Some(stopwatchOutline)
  | "globe" => Some(globe)
  | "globe-outline" => Some(globeOutline)
  | "suitcase" => Some(suitcase)
  | "suitcase-outline" => Some(suitcaseOutline)
  | "level" => Some(level)
  | "level-outline" => Some(levelOutline)
  | "share-1" => Some(share1)
  | "share-2" => Some(share2)
  | "restart-outline" => Some(restartOutline)
  | "browser-outline" => Some(browserOutline)
  | _ => None
  };

let emptyStyle = ReactNative.Style.style();

[@react.component]
let make =
    (
      ~icon,
      ~size=24.0,
      ~color=?,
      ~wrapperStyle=emptyStyle,
      ~theme=Hero_Theme.default,
    ) => {
  let iconColor =
    color->Belt.Option.getWithDefault @@ getColorProperty @@  theme##icon##icon;

  switch (xmlFromIcon(icon)) {
  | None => ReasonReact.null
  | Some(xml) =>
    <View
      style={StyleSheet.flatten([|
        Style.(
          style(
            ~display=`flex,
            ~alignItems=`center,
            ~justifyContent=`center,
            ~width=size->dp,
            ~height=size->dp,
            (),
          )
        ),
        wrapperStyle,
      |])}>
      <SvgXml
        xml
        override={
          "width": size,
          "height": size,
          "stroke": iconColor,
          "fill": iconColor,
        }
      />
    </View>
  };
};

let default = Helpers.injectTheme(make);
