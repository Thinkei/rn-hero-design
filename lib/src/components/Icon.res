open ReactNative

@get external getColorProperty: Style.t => Color.t = "color"

module SvgXml = {
  @react.component @module("react-native-svg")
  external make: (~xml: string, ~override: 'a) => React.element = "SvgXml"
}

@module("../icons/email") external email: string = "default"
@module("../icons/email-outline")
external emailOutline: string = "default"
@module("../icons/eye") external eye: string = "default"
@module("../icons/eye-outline") external eyeOutline: string = "default"
@module("../icons/eye-invisible")
external eyeInvisible: string = "default"
@module("../icons/eye-invisible-outline")
external eyeInvisibleOutline: string = "default"
@module("../icons/ok-circle") external okCircle: string = "default"
@module("../icons/calendar") external calendar: string = "default"
@module("../icons/clock-circle-outline")
external clockCircleOutline: string = "default"
@module("../icons/comment-outline")
external commentOutline: string = "default"
@module("../icons/cancel-outline")
external cancelOutline: string = "default"
@module("../icons/plus-circle-outline")
external plusCircleOutline: string = "default"
@module("../icons/plus-outline") external plusOutline: string = "default"
@module("../icons/piggy-bank-outline")
external piggyBankOutline: string = "default"
@module("../icons/target-outline")
external targetOutline: string = "default"
@module("../icons/single-right-outline")
external singleRightOutline: string = "default"
@module("../icons/more-vertical")
external moreVertical: string = "default"
@module("../icons/calendar-outline")
external calendarOutline: string = "default"
@module("../icons/phone-outline")
external phoneOutline: string = "default"
@module("../icons/face-id") external faceId: string = "default"
@module("../icons/single-left-outline")
external singleLeftOutline: string = "default"
@module("../icons/arrow-down") external arrowDown: string = "default"
@module("../icons/arrow-up") external arrowUp: string = "default"
@module("../icons/camera-outline")
external cameraOutline: string = "default"
@module("../icons/paperclip") external paperclip: string = "default"
@module("../icons/image-outline")
external imageOutline: string = "default"
@module("../icons/file-outline") external fileOutline: string = "default"
@module("../icons/warning-circle-outline")
external warningCircleOutline: string = "default"
@module("../icons/warning-circle")
external warningCircle: string = "default"
@module("../icons/radio-active") external radioActive: string = "default"
@module("../icons/radio-inactive")
external radioInactive: string = "default"
@module("../icons/plus-circle") external plusCircle: string = "default"
@module("../icons/cancel-circle")
external cancelCircle: string = "default"
@module("../icons/cancel-circle-outline")
external cancelCircleOutline: string = "default"
@module("../icons/ok-circle-outline")
external okCircleOutline: string = "default"
@module("../icons/folder") external folder: string = "default"
@module("../icons/folder-outline")
external folderOutline: string = "default"
@module("../icons/stopwatch") external stopwatch: string = "default"
@module("../icons/stopwatch-outline")
external stopwatchOutline: string = "default"
@module("../icons/globe") external globe: string = "default"
@module("../icons/globe-outline")
external globeOutline: string = "default"
@module("../icons/suitcase") external suitcase: string = "default"
@module("../icons/suitcase-outline")
external suitcaseOutline: string = "default"
@module("../icons/level") external level: string = "default"
@module("../icons/level-outline")
external levelOutline: string = "default"
@module("../icons/user") external user: string = "default"
@module("../icons/user-outline") external userOutline: string = "default"
@module("../icons/multiple-stars")
external multipleStars: string = "default"
@module("../icons/multiple-users")
external multipleUsers: string = "default"
@module("../icons/remove-circle")
external removeCircle: string = "default"
@module("../icons/remove-circle-outline")
external removeCircleOutline: string = "default"
@module("../icons/bell-outline") external bellOutline: string = "default"
@module("../icons/health-bag-outline")
external healthBagOutline: string = "default"
@module("../icons/plane-outline")
external planeOutline: string = "default"
@module("../icons/recognition-medal-outline")
external recognitionMedalOutline: string = "default"
@module("../icons/talk-outline") external talkOutline: string = "default"
@module("../icons/speaker-outline")
external speakerOutline: string = "default"
@module("../icons/home-outline") external homeOutline: string = "default"
@module("../icons/more-horizontal")
external moreHorizontal: string = "default"
@module("../icons/cog-outline") external cogOutline: string = "default"
@module("../icons/search-outline")
external searchOutline: string = "default"
@module("../icons/speaker") external speaker: string = "default"
@module("../icons/raising-hands")
external raisingHands: string = "default"
@module("../icons/certificate") external certificate: string = "default"
@module("../icons/recognition-medal")
external recognitionMedal: string = "default"
@module("../icons/birthday-cake")
external birthdayCake: string = "default"
@module("../icons/remove") external remove: string = "default"
@module("../icons/add") external add: string = "default"
@module("../icons/checkmark") external checkmark: string = "default"
@module("../icons/coins-outline")
external coinsOutline: string = "default"
@module("../icons/number") external number: string = "default"
@module("../icons/percentage") external percentage: string = "default"

@module("../icons/format_bold") external formatBold: string = "default"
@module("../icons/format_italic")
external formatItalic: string = "default"
@module("../icons/format_underlined")
external formatUnderlined: string = "default"
@module("../icons/format_list_bulleted")
external formatListBulleted: string = "default"
@module("../icons/format_list_numbered")
external formatListNumbered: string = "default"
@module("../icons/looks_one") external looksOne: string = "default"
@module("../icons/looks_two") external looksTwo: string = "default"

let xmlFromIcon = icon =>
  switch icon {
  | "email" => Some(email)
  | "email-outline" => Some(emailOutline)
  | "eye" => Some(eye)
  | "eye-outline" => Some(eyeOutline)
  | "eye-invisible" => Some(eyeInvisible)
  | "eye-invisible-outline" => Some(eyeInvisibleOutline)
  | "ok-circle" => Some(okCircle)
  | "calendar" => Some(calendar)
  | "calendar-outline" => Some(calendarOutline)
  | "clock-circle-outline" => Some(clockCircleOutline)
  | "comment-outline" => Some(commentOutline)
  | "cancel-outline" => Some(cancelOutline)
  | "plus-circle-outline" => Some(plusCircleOutline)
  | "plus-outline" => Some(plusOutline)
  | "piggy-bank-outline" => Some(piggyBankOutline)
  | "target-outline" => Some(targetOutline)
  | "single-right-outline" => Some(singleRightOutline)
  | "single-left-outline" => Some(singleLeftOutline)
  | "more-vertical" => Some(moreVertical)
  | "phone-outline" => Some(phoneOutline)
  | "face-id" => Some(faceId)
  | "arrow-down" => Some(arrowDown)
  | "arrow-up" => Some(arrowUp)
  | "camera-outline" => Some(cameraOutline)
  | "paperclip" => Some(paperclip)
  | "image-outline" => Some(imageOutline)
  | "file-outline" => Some(fileOutline)
  | "warning-circle-outline" => Some(warningCircleOutline)
  | "warning-circle" => Some(warningCircle)
  | "radio-active" => Some(radioActive)
  | "radio-inactive" => Some(radioInactive)
  | "format_bold" => Some(formatBold)
  | "format_italic" => Some(formatItalic)
  | "format_underlined" => Some(formatUnderlined)
  | "format_list_numbered" => Some(formatListNumbered)
  | "format_list_bulleted" => Some(formatListBulleted)
  | "looks_one" => Some(looksOne)
  | "looks_two" => Some(looksTwo)
  | "plus-circle" => Some(plusCircle)
  | "cancel-circle" => Some(cancelCircle)
  | "cancel-circle-outline" => Some(cancelCircleOutline)
  | "ok-circle-outline" => Some(okCircleOutline)
  | "folder" => Some(folder)
  | "folder-outline" => Some(folderOutline)
  | "stopwatch" => Some(stopwatch)
  | "stopwatch-outline" => Some(stopwatchOutline)
  | "globe" => Some(globe)
  | "globe-outline" => Some(globeOutline)
  | "suitcase" => Some(suitcase)
  | "suitcase-outline" => Some(suitcaseOutline)
  | "level" => Some(level)
  | "level-outline" => Some(levelOutline)
  | "user" => Some(user)
  | "user-outline" => Some(userOutline)
  | "multiple-stars" => Some(multipleStars)
  | "multiple-users" => Some(multipleUsers)
  | "remove-circle" => Some(removeCircle)
  | "remove-circle-outline" => Some(removeCircleOutline)
  | "bell-outline" => Some(bellOutline)
  | "health-bag-outline" => Some(healthBagOutline)
  | "plane-outline" => Some(planeOutline)
  | "recognition-medal-outline" => Some(recognitionMedalOutline)
  | "talk-outline" => Some(talkOutline)
  | "speaker-outline" => Some(speakerOutline)
  | "home-outline" => Some(homeOutline)
  | "more-horizontal" => Some(moreHorizontal)
  | "cog-outline" => Some(cogOutline)
  | "search-outline" => Some(searchOutline)
  | "speaker" => Some(speaker)
  | "raising-hands" => Some(raisingHands)
  | "certificate" => Some(certificate)
  | "recognition-medal" => Some(recognitionMedal)
  | "birthday-cake" => Some(birthdayCake)
  | "remove" => Some(remove)
  | "add" => Some(add)
  | "checkmark" => Some(checkmark)
  | "coins-outline" => Some(coinsOutline)
  | "number" => Some(number)
  | "percentage" => Some(percentage)
  | _ => None
  }

let emptyStyle = ReactNative.Style.style()

@genType @react.component
let make = (
  ~icon,
  ~size=24.0,
  ~color=?,
  ~wrapperStyle=emptyStyle,
  ~theme=Hero_Theme.default,
  ~testID="",
) => {
  let iconColor = \"@@"(
    color->Belt.Option.getWithDefault,
    \"@@"(getColorProperty, theme["icon"]["icon"]),
  )

  switch xmlFromIcon(icon) {
  | None => React.null
  | Some(xml) =>
    <View
      testID
      style={StyleSheet.flatten([
        {
          open Style
          style(
            ~display=#flex,
            ~alignItems=#center,
            ~justifyContent=#center,
            ~width=size->dp,
            ~height=size->dp,
            (),
          )
        },
        wrapperStyle,
      ])}>
      <SvgXml
        xml
        override={
          "width": size,
          "height": size,
          "stroke": iconColor,
          "fill": iconColor,
        }
      />
    </View>
  }
}

@genType
let default = Helpers.injectTheme(make)
