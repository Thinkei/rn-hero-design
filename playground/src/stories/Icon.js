import React, { useMemo } from 'react';
import { Icon, Container, injectTheme } from 'rn-hero-design';

const filledIcons = [
  'ok-circle',
  'cancel-circle',
  'plus-circle',
  'warning-circle',
  'email',
  'eye',
  'eye-invisible',
  'folder',
  'stopwatch',
  'calendar',
  'globe',
  'suitcase',
  'level',

  'more-vertical',
  'radio-active',
  'radio-inactive',
];

const outlinedIcons = [
  'ok-circle-outline',
  'cancel-circle-outline',
  'plus-circle-outline',
  'warning-circle-outline',
  'email-outline',
  'eye-outline',
  'eye-invisible-outline',
  'folder-outline',
  'stopwatch-outline',
  'calendar-outline',
  'globe-outline',
  'suitcase-outline',
  'level-outline',

  'plus-outline',
  'cancel-outline',
  'single-right-outline',
  'single-left-outline',
  'arrow-down',
  'clock-circle-outline',
  'comment-outline',
  'piggy-bank-outline',
  'target-outline',
  'phone-outline',
  'face-id',
  'camera-outline',
  'paperclip',
  'image-outline',
  'file-outline',
];

const IconScreen = ({ theme }) => {
  const iconStyle = useMemo(
    () => ({
      width: theme.variables.LARGE_SIZE * 2,
      height: theme.variables.LARGE_SIZE * 2,
    }),
    [theme],
  );

  return (
    <>
      <Container direction="row" style={{ flexWrap: 'wrap' }}>
        {filledIcons.map((icon) => (
          <Icon
            key={icon}
            icon={icon}
            size={24}
            color={theme.variables.FOCUS_BLUE_1}
            wrapperStyle={iconStyle}
          />
        ))}
      </Container>

      <Container direction="row" style={{ flexWrap: 'wrap' }}>
        {outlinedIcons.map((icon) => (
          <Icon
            key={icon}
            icon={icon}
            size={24}
            color={theme.variables.FOCUS_BLUE_1}
            wrapperStyle={iconStyle}
          />
        ))}
      </Container>
    </>
  );
};

export default injectTheme(IconScreen);
