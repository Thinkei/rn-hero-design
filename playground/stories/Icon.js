import React from 'react';
import { StyleSheet } from 'react-native';
import { Icon, Container } from 'rn-hero-design';

const IconScreen = () => (
  <>
    <Container direction="row" style={styles.container}>
      <Icon icon="email" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon icon="eye" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon icon="eye-invisible" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon icon="ok-circle" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon icon="calendar" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon icon="more-vertical" size={24} wrapperStyle={styles.iconWrapper} />
    </Container>
    <Container direction="row" style={styles.container}>
      <Icon icon="email-outline" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon icon="eye-outline" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon
        icon="eye-invisible-outline"
        size={24}
        wrapperStyle={styles.iconWrapper}
      />
      <Icon icon="cancel-outline" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon
        icon="clock-circle-outline"
        size={24}
        wrapperStyle={styles.iconWrapper}
      />
      <Icon
        icon="comment-outline"
        size={24}
        wrapperStyle={styles.iconWrapper}
      />
      <Icon
        icon="plus-circle-outline"
        size={24}
        wrapperStyle={styles.iconWrapper}
      />
      <Icon icon="plus-outline" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon
        icon="piggy-bank-outline"
        size={24}
        wrapperStyle={styles.iconWrapper}
      />
      <Icon icon="target-outline" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon
        icon="single-right-outline"
        size={24}
        wrapperStyle={styles.iconWrapper}
      />
      <Icon
        icon="calendar-outline"
        size={24}
        wrapperStyle={styles.iconWrapper}
      />
      <Icon icon="phone-outline" size={24} wrapperStyle={styles.iconWrapper} />
      <Icon icon="face-id" size={24} wrapperStyle={styles.iconWrapper} />
    </Container>
  </>
);

const styles = StyleSheet.create({
  container: {
    flexWrap: 'wrap',
    paddingBottom: 0,
  },
  iconWrapper: {
    width: 48,
    height: 48,
  },
});

export default IconScreen;
