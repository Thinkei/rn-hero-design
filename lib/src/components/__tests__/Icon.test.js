import React from 'react';

import { render } from '@testing-library/react-native';

import Icon from '../Icon.gen';

describe('Icon', () => {
  it.each`
    element
    ${<Icon icon="email-outline" />}
    ${<Icon icon="email-outline" color="red" />}
    ${<Icon icon="email-outline" color="red" size={48} />}
  `('renders correctly', async ({ element }) => {
    const { baseElement } = render(element);

    expect(baseElement).toMatchSnapshot();
  });
});
