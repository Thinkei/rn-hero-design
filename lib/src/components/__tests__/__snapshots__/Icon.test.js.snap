// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 24,
      "justifyContent": "center",
      "width": 24,
    }
  }
>
  <View
    override={
      Object {
        "color": "#000000",
        "height": 24,
        "width": 24,
      }
    }
    xml={
      Object {
        "testUri": "../../../lib/src/icons/email-outline.svg",
      }
    }
  />
</View>
`;

exports[`Icon renders correctly 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 24,
      "justifyContent": "center",
      "width": 24,
    }
  }
>
  <View
    override={
      Object {
        "color": "red",
        "height": 24,
        "width": 24,
      }
    }
    xml={
      Object {
        "testUri": "../../../lib/src/icons/email-outline.svg",
      }
    }
  />
</View>
`;

exports[`Icon renders correctly 3`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 48,
      "justifyContent": "center",
      "width": 48,
    }
  }
>
  <View
    override={
      Object {
        "color": "red",
        "height": 48,
        "width": 48,
      }
    }
    xml={
      Object {
        "testUri": "../../../lib/src/icons/email-outline.svg",
      }
    }
  />
</View>
`;
